typescript/src/api-doc-types/PropertyMemberDoc.ts

```diff
@@ -1,9 +1,34 @@
+import { Declaration, GetAccessorDeclaration, SetAccessorDeclaration, SignatureDeclaration, Symbol, SyntaxKind } from 'typescript';
+import { getDeclarationTypeText } from "../services/TsParser/getDeclarationTypeText";
+import { getParameters } from '../services/TsParser/getParameters';
 import { encodeAnchor } from '../utils/encodeAnchor';
+import { AccessorInfoDoc } from './AccessorInfoDoc';
+import { ContainerExportDoc } from './ContainerExportDoc';
 import { MemberDoc } from './MemberDoc';
 
 export class PropertyMemberDoc extends MemberDoc {
   name = this.symbol.name;
   anchor = encodeAnchor(this.name);
   id = `${this.containerDoc.id}.${this.name}`;
   aliases = this.containerDoc.aliases.map(alias => `${alias}.${this.name}` );
+  getAccessor: AccessorInfoDoc | null;
+  setAccessor: AccessorInfoDoc | null;
+
+  constructor(
+    containerDoc: ContainerExportDoc,
+    symbol: Symbol,
+    declaration: Declaration | null,
+    getAccessorDeclaration: GetAccessorDeclaration | null,
+    setAccessorDeclaration: SetAccessorDeclaration | null,
+    basePath: string,
+    namespacesToInclude: string[],
+    isStatic: boolean,
+  ) {
+    super(containerDoc, symbol, (declaration || getAccessorDeclaration || setAccessorDeclaration)!, basePath, namespacesToInclude, isStatic);
+
+    // If this property has accessors then compute the type based on that instead
+    this.getAccessor = getAccessorDeclaration && new AccessorInfoDoc('get', this, getAccessorDeclaration, basePath, namespacesToInclude);
+    this.setAccessor = setAccessorDeclaration && new AccessorInfoDoc('set', this, setAccessorDeclaration, basePath, namespacesToInclude);
+    this.type = this.type || this.setAccessor && this.setAccessor.parameters[0].split(/:\s?/)[1] || '';
```

Comment:
If we have passed in `declaration` or `getAccessorDeclaration` then we will have extracted the type from one of those, if possible.

Reply:
Yeah that's true. On Material 2 we have a small workaround for this, but it still uses the declaration to retrieve the type. See [here](https://github.com/angular/material2/blob/master/tools/dgeni/common/dgeni-accessors-parse.ts)

Reply:
Might be worth to also check how the description of the getter/setter is being handled. I noticed that the description is only being read from one place.

```ts

get test() {}

/** Description on the setter*/
set test(val: boolean) {}

```

Reply:
So the getter and setter are docs in their own right, so their documentation should be processed for jsdoc tags, etc. You can then access them via the `member.getAccessor` and `member.setAccessor` for use in the property template.

This is similar to how overloads are being dealt with. With overloads we extract the content of the "actual" function for the member doc but then attach the content for each of the overload declarations to the overloads for that member.

In the case of a property with a getter and setter, we extract the content from the first declaration that is defined from `[declaration, getter, setter]`. It would seem to me more common to put the documentation (if there is only one) on the getter not the setter YMMV. But if you do put it only on the setter then you could easily get around this by doing something like:

Something like:

```
<h2 class="property-name">{$ member.name $}</h2>
<p class="description">{$ member.content or member.setAccessor and member.getAccessor.content $}</p>
```

Is this too unreasonable? Do you really put documentation on the setter only?

Reply:
Not really, on Material I remember seeing accessors that have the descriptions on a `setter` or on the `getter` and I want to make sure we don't need to enforce something here.

For the description, for me it makes sense to have a `description` property on the "MemberDoc". The description there should be retrieved in a similar way as the type, because it can be either on the getter or setter.

```ts
memberDoc.description
```

Reply:
OK, I'll transfer that from the relevant accessor too.

Reply:
Done PTAL

Reply:
Just tested it locally and the types seem to work fine now. For the descriptions, I'm not sure what the different to `content` is here? When debugging most of our accessors in Material, they don't seem to have a `content` either.

Isn't there a way to set the `description` as well, based on the two cases I described above?

```ts
    // This works fine now. Seems like every thing has a type now!
    if (!propertyDoc.type) {
      console.log('No type for', propertyDoc.name);
    }

    // Likely every accessor doesn't have a description
    if (!propertyDoc.description) {
      console.log('No description for', propertyDoc.name, propertyDoc.content);
    }
```

This is the code-snippet we currently use in Material to receive the proper description for an accessor.

https://github.com/angular/material2/blob/4d97271786855598cfd18921641d5ffaafa079d8/tools/dgeni/common/dgeni-accessors-parse.ts#L51-L54

Reply:
Remember that this processor runs before the jsdic processing happens. The content field contains the raw content. After jsdoc runs there will be a description property too

Reply:
Hmm I'm not sure why the `content` is empty then. For me likely every accessor document has a `content` that is empty.